optuna/samplers/_motpe.py

```diff
@@ -0,0 +1,627 @@
+import math
+from typing import Any
+from typing import Callable
+from typing import cast
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import numpy as np
+
+import optuna
+from optuna import distributions
+from optuna.distributions import BaseDistribution
+from optuna.samplers._random import RandomSampler
+from optuna.samplers._tpe.parzen_estimator import _ParzenEstimator
+from optuna.samplers._tpe.parzen_estimator import _ParzenEstimatorParameters
+from optuna.samplers._tpe.sampler import TPESampler
+from optuna.study import StudyDirection
+
+
+EPS = 1e-12
+_SPLITCACHE_KEY = "motpe:splitcache"
+_WEIGHTS_BELOW_KEY = "motpe:weights_below"
+
+
+def default_gamma(x: int) -> int:
+    return int(np.floor(0.1 * x))
+
+
+def _default_weights_above(x: int) -> np.ndarray:
+    return np.ones(x)
+
+
+class MOTPESampler(TPESampler):
```

Comment:
Would just like to verify but this sampler is stable and not experimental? C.f. https://github.com/optuna/optuna/commit/909324a09d8a45f07b78422284f04cc8c4ca2349

Reply:
@HideakiImamura , @toshihikoyanase you probably know better than myself given #1530. Do you have any comments on this?

Reply:
I'm not pretty sure, but MoTPE may be able to be merged with TPE. Such a change may affect the public interface, and I think it is reasonable to keep the MoTPE experimental.

Reply:
I agree with @toshihikoyanase. As a follow-up task, we can merge the multi-objective TPE with TPE after the v2.4.0 release.